litr/src/main/java/com/linkedin/android/litr/utils/TranscoderUtils.java

```diff
@@ -133,34 +132,35 @@ public static int estimateVideoTrackBitrate(@NonNull MediaSource mediaSource, in
             return videoTrackFormat.getInteger(MediaFormat.KEY_BIT_RATE);
         }
 
-        long unallocatedSize = mediaSource.getSize();
-        long totalPixels = 0;
+        float unallocatedSize = mediaSource.getSize();
+        float totalPixels = 0;
         int trackCount = mediaSource.getTrackCount();
         for (int track = 0; track < trackCount; track++) {
             MediaFormat trackFormat = mediaSource.getTrackFormat(track);
             if (trackFormat.containsKey(MediaFormat.KEY_MIME)) {
                 if (trackFormat.containsKey(MediaFormat.KEY_BIT_RATE) && trackFormat.containsKey(MediaFormat.KEY_DURATION)) {
                     int bitrate = trackFormat.getInteger(MediaFormat.KEY_BIT_RATE);
                     long duration = trackFormat.getLong(MediaFormat.KEY_DURATION);
-                    unallocatedSize -= bitrate * TimeUnit.MICROSECONDS.toSeconds(duration) / 8;
+                    unallocatedSize -= bitrate * TimeUtils.microsToSeconds(duration) / BITS_IN_BYTE;
                 } else {
                     String mimeType = trackFormat.getString(MediaFormat.KEY_MIME);
                     if (mimeType.startsWith("video")) {
                         totalPixels += trackFormat.getInteger(MediaFormat.KEY_WIDTH)
                             * trackFormat.getInteger(MediaFormat.KEY_HEIGHT)
-                            * TimeUnit.MICROSECONDS.toSeconds(trackFormat.getLong(MediaFormat.KEY_DURATION));
+                            * TimeUtils.microsToSeconds(trackFormat.getLong(MediaFormat.KEY_DURATION));
                     }
                 }
             }
         }
 
-        long trackPixels = videoTrackFormat.getInteger(MediaFormat.KEY_WIDTH)
+        float videoTrackDuration = TimeUtils.microsToSeconds(videoTrackFormat.getLong(MediaFormat.KEY_DURATION));
+        float trackPixels = videoTrackFormat.getInteger(MediaFormat.KEY_WIDTH)
             * videoTrackFormat.getInteger(MediaFormat.KEY_HEIGHT)
-            * TimeUnit.MICROSECONDS.toSeconds(videoTrackFormat.getLong(MediaFormat.KEY_DURATION));
+            * videoTrackDuration;
 
-        long trackSize = unallocatedSize * trackPixels / totalPixels;
-
-        return (int) (trackSize * 8 / TimeUnit.MICROSECONDS.toSeconds(videoTrackFormat.getLong(MediaFormat.KEY_DURATION)));
+        float trackSize = totalPixels > 0 ? unallocatedSize * trackPixels / totalPixels : unallocatedSize;
+        videoTrackDuration = videoTrackDuration == 0 ? 1 : videoTrackDuration;
```

Comment:
I don't think this really applies anymore, since we are working with <1 fractional values now. Let's guard in a different way: 
// zero duration means empty track
if (videoTrackDuration == 0) {
    return 0; 
}

we can actually do this early and return early if duration is zero, since this means that track has no pixels in it.

Reply:
Fixed